packages/controllers/src/idCounter.ts

```diff
@@ -0,0 +1,5 @@
+let id = 1;
```

Comment:
starting at `0` encountered many bugs. should maybe use `uuid` here as well? its already used in this package

Reply:
For IDs of this nature, I've always been a fan of the `getUniqueId` implementation in `json-rpc-engine`: https://github.com/MetaMask/json-rpc-engine/blob/9b04c79/src/getUniqueId.ts/#L1-L9

Otherwise, I prefer `nanoid` and then finally `uuid`. `nanoid` is faster and more concise than `uuid/v4` for practically equivalent randomness.